vita3k/modules/SceGxm/SceGxm.cpp

```diff
@@ -3145,19 +3145,15 @@ EXPORT(int, sceGxmProgramCheck, const SceGxmProgram *program) {
 
 EXPORT(Ptr<SceGxmProgramParameter>, sceGxmProgramFindParameterByName, const SceGxmProgram *program, const char *name) {
     TRACY_FUNC(sceGxmProgramFindParameterByName, program, name);
-    const MemState &mem = emuenv.mem;
     assert(program);
     if (!program || !name)
         return Ptr<SceGxmProgramParameter>();
 
-    const SceGxmProgramParameter *const parameters = reinterpret_cast<const SceGxmProgramParameter *>(reinterpret_cast<const uint8_t *>(&program->parameters_offset) + program->parameters_offset);
+    auto parameters = program->program_parameters();
     for (uint32_t i = 0; i < program->parameter_count; ++i) {
-        const SceGxmProgramParameter *const parameter = &parameters[i];
-        const uint8_t *const parameter_bytes = reinterpret_cast<const uint8_t *>(parameter);
-        const char *const parameter_name = reinterpret_cast<const char *>(parameter_bytes + parameter->name_offset);
-        if (strcmp(parameter_name, name) == 0) {
-            const Address parameter_address = static_cast<Address>(parameter_bytes - &mem.memory[0]);
-            return Ptr<SceGxmProgramParameter>(parameter_address);
+        auto parameter = &parameters[i];
+        if (strcmp(parameter->name(), name) == 0) {
+            return Ptr<SceGxmProgramParameter>(parameter, emuenv.mem);
```

Comment:
Really nice simplification. However I realize that Ptr<SceGxmProgramParameter>(parameter, emuenv.mem) can cause issues when using a page table (parameter_bytes - &mem.memory[0] is wrong for mapped memory which is likely the case here). 

I think an easy way to solve it would be to get the program as a Ptr: Ptr<const SceGxmProgram> program_ptr.

Then you can return 
(program_ptr.cast<uint8_t>() + program->parameters_offset + i*sizeof(SceGxmProgramParameter)).cast<SceGxmProgramParameter>();